<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1106</width>
    <height>473</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Path Planning Simulator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>10</y>
      <width>641</width>
      <height>401</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout"/>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>665</x>
      <y>0</y>
      <width>201</width>
      <height>196</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox::title{
	color :  black;
	subcontrol-origin: margin;
   left: 10px;
   padding: 0 3px 0 3px;
}

QGroupBox{
	border : 2px solid gray;
	border-radius : 5px;
	margin-top: 0.5em;
}</string>
    </property>
    <property name="title">
     <string>Initialize Data</string>
    </property>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>181</width>
       <height>46</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Robot Position</string>
     </property>
     <widget class="QLineEdit" name="Edit_RPosition_x">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>15</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_RPosition_y">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>15</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>26</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>X :</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>95</x>
        <y>20</y>
        <width>26</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Y :</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>181</width>
       <height>46</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Ball Position</string>
     </property>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>95</x>
        <y>20</y>
        <width>26</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Y :</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_BallPosition_y">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>15</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_BallPosition_x">
      <property name="geometry">
       <rect>
        <x>35</x>
        <y>15</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>26</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>X :</string>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="Button_Generate_Data">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>160</y>
       <width>181</width>
       <height>26</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Generate Initialize Data</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>135</y>
       <width>56</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Obstacle</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_ObstacleNumber">
     <property name="geometry">
      <rect>
       <x>75</x>
       <y>135</y>
       <width>66</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>665</x>
      <y>210</y>
      <width>431</width>
      <height>201</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox::title{
	color :  black;
	subcontrol-origin: margin;
   left: 10px;
   padding: 0 3px 0 3px;
}

QGroupBox{
	border : 2px solid gray;
	border-radius : 5px;
	margin-top: 0.5em;
}</string>
    </property>
    <property name="title">
     <string>Path Planning</string>
    </property>
    <widget class="QGroupBox" name="groupBox_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>15</y>
       <width>186</width>
       <height>176</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Genetic Algorithm Parameter</string>
     </property>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>15</y>
        <width>86</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Gen Number</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>40</y>
        <width>86</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Population</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>65</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>CrossOver Rate</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>90</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Mutation Rate</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>115</y>
        <width>91</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Generation</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_GA_Gen">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>15</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_GA_Population">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>40</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_GA_CO_Rate">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>65</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_GA_MU_Rate">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>90</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="Edit_GA_Generation">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>115</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="Button_Generate_GA_Parameter">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>140</y>
        <width>156</width>
        <height>26</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Generate GA Parameter</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>15</y>
       <width>211</width>
       <height>56</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="title">
      <string>A star path planning</string>
     </property>
     <widget class="QPushButton" name="Button_Generate_Astar">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>20</y>
        <width>181</width>
        <height>26</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Generate A* Algorithm</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>80</y>
       <width>211</width>
       <height>111</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="title">
      <string>Path Planning Result</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_8">
    <property name="geometry">
     <rect>
      <x>885</x>
      <y>0</y>
      <width>206</width>
      <height>196</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>9</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox::title{
	color :  black;
	subcontrol-origin: margin;
   left: 10px;
   padding: 0 3px 0 3px;
}

QGroupBox{
	border : 2px solid gray;
	border-radius : 5px;
	margin-top: 0.5em;
}</string>
    </property>
    <property name="title">
     <string>Initialize via Point</string>
    </property>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>15</x>
       <y>20</y>
       <width>101</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>viaPoint Number</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spinBox_NodeNumber">
     <property name="geometry">
      <rect>
       <x>125</x>
       <y>20</y>
       <width>66</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="Button_Generate_viaPoint">
     <property name="geometry">
      <rect>
       <x>15</x>
       <y>45</y>
       <width>176</width>
       <height>26</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>9</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Generate Initialize viaPoint</string>
     </property>
    </widget>
    <widget class="QListWidget" name="List_viaPoint">
     <property name="geometry">
      <rect>
       <x>15</x>
       <y>75</y>
       <width>176</width>
       <height>96</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
    </widget>
    <widget class="QLabel" name="Label_Execution_viaPoint">
     <property name="geometry">
      <rect>
       <x>15</x>
       <y>175</y>
       <width>176</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>8</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Execution Time 0 ms</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1106</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="font">
    <font>
     <pointsize>9</pointsize>
    </font>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
